packages/core/src/indexing/query-filter-converter.ts

```diff
@@ -0,0 +1,224 @@
+import { Knex } from 'knex'
+import { NonBooleanValueFilterType, ObjectFilter, QueryFilters } from './query-filter-parser.js'
+
+export const DATA_FIELD = 'stream_content'
+
+type DBQuery = Knex.QueryBuilder
+type WhereFunc = (DBQuery) => DBQuery
+
+enum Combinator {
+  And,
+  Or,
+}
+
+type ConversionState<T> = {
+  filter: T
+  combinator?: Combinator
+  negated: boolean
+}
+
+function nextState<Old, New>(state: ConversionState<Old>, next: New): ConversionState<New> {
+  return {
+    filter: next,
+    negated: state.negated,
+    combinator: state.combinator,
+  }
+}
+
+function handleQuery(
```

Comment:
would be really nice to have some comments on this explaining the different args and how they are meant to be used